simplq/src/store/asyncActions/getToken.js

```diff
@@ -0,0 +1,30 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import useAuth, { makeAuthedRequest } from 'api/auth';
+import * as RequestFactory from 'api/requestFactory';
+import { useCallback } from 'react';
+
+const typePrefix = 'getToken/action';
+
+/**
+ * A hook to access the getToken async action creator.
+ *
+ * @returns — getToken async action creator
+ */
+const useGetToken = () => {
+  const auth = useAuth();
+
+  const getToken = useCallback(
```

Comment:
@mradenovic Shall we move the useCallback into the action creator hook? Or is there a good reason to use it when called?

Reply:
`useCallback` should be used only if an action is dispatched from the `useEffect`. Look at the Join page for an example.

Reply:
Is there a problem if we use it everytime?

Reply:
To be honest, I am not sure, I tried to research with no final conclusion. I think every `useCallback` call will grab some memory. And I don't want to pollute async actions with an unnecessary wrapper.